mev-relay-rs/src/relay.rs

```diff
@@ -62,22 +74,30 @@ fn validate_signed_block(
     signed_block: &mut SignedBlindedBeaconBlock,
     public_key: &BlsPublicKey,
     local_payload: &mut ExecutionPayload,
+    current_slot: Slot,
+    genesis_validators_root: Root,
+
     context: &Context,
 ) -> Result<(), Error> {
     let local_block_hash = local_payload.block_hash();
     let block_hash = signed_block.block_hash();
+    let slot = signed_block.slot();
     if block_hash != local_block_hash {
         return Err(Error::UnknownBlock)
     }
 
     // OPTIONAL:
     // -- verify w/ consensus?
     // verify slot is timely
-    // verify proposer_index is correct
+    if slot < current_slot || slot > current_slot + PROPOSAL_TOLERANCE_DELAY {
+        return Err(Error::UnknownBlock);
+    }
```

Comment:
if a bid comes in at the end of the slot we could get to a case where the bid races our update of time

if anything I'd rather convert the current slot to a timestamp and allow for some propagation tolerance, idk 500ms, to enforce any time deadlines here